service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java

```diff
@@ -92,6 +99,46 @@ public void deleteLandingZone(String landingZoneId) {
     landingZoneService.deleteLandingZone(landingZoneId);
   }
 
+  public ApiAzureLandingZoneResourcesList listAzureLandingZoneResources(String landingZoneId) {
+    features.azureEnabledCheck();
+    LandingZoneResourcesByPurpose groupedResources =
+        landingZoneService.listResourcesWithPurposes(landingZoneId);
+
+    var result = new ApiAzureLandingZoneResourcesList().id(landingZoneId);
+
+    groupedResources
+        .deployedResources()
+        .forEach(
+            (p, dp) ->
+                result.addResourcesItem(
+                    new ApiAzureLandingZoneResourcesPurposeGroup()
+                        .purpose(p.getClass().getSimpleName())
+                        .deployedResources(
+                            dp.stream()
+                                .map(r -> toApiAzureLandingZoneDeployedResource(r, p))
+                                .toList())));
+    return result;
+  }
+
+  private ApiAzureLandingZoneDeployedResource toApiAzureLandingZoneDeployedResource(
+      LandingZoneResource resource, LandingZonePurpose purpose) {
+    if (purpose.getClass().equals(ResourcePurpose.class)) {
+      return new ApiAzureLandingZoneDeployedResource()
+          .resourceId(resource.resourceId())
+          .resourceType(resource.resourceType())
+          .region(resource.region());
+    }
+    if (purpose.getClass().equals(SubnetResourcePurpose.class)) {
+      return new ApiAzureLandingZoneDeployedResource()
+          .resourceParentId(resource.resourceParentId().get()) // Only available for subnets
+          .resourceName(resource.resourceName().get()) // Only available for subnets
+          .resourceType(resource.resourceType())
+          .region(resource.region());
+    }
+    throw new LandingZoneUnsupportedPurposeException(
```

Comment:
Why is this exception a 400 BadRequest? The caller didn't make a bad request. The return value from the landing zone library is invalid as far as this code is concerned. I think it should either be not-implemented or a 500 server error.

Reply:
OK, not-implemented is logical.